### PR TITLE
fix(sw): handle dev serviceworker 404

### LINKED ISSUE #64: Prevent request for sw when in non-sw development
> The ENOENT error for the serviceworker (and the corresponding 404 in your browser) is a known issue, and behaving more or less as expected. We don't want to install a service worker by default in developer mode unless you're working on the service worker itself. The 404 is unsightly, though, and we'll fix that by making the client side aware that it shouldn't request one.

[Context](https://github.com/magento-research/venia-pwa-concept/issues/56#issuecomment-386669379)

### PR DESCRIPTION
**Pull request by zetlen**
Modified `src/index.js` to only register a service worker if `process.env.SERVICE_WORKER_FILE_NAME` is truthy.

Modified `webpack.config.js` to set `process.env.SERVICE_WORKER_FILE_NAME` to `false` in development mode, unless the special `ENABLE_SERVICE_WORKER_DEBUGGING` env var is truthy.

Now, service workers are only generated OR requested:
 - in production
 - when they are directly under development, indicated by the env var `ENABLE_SERVICE_WORKER_DEBUGGING`.

Fixes #64.

### CODE REVIEW

File: theme-frontend-venia/webpack.config.js [hunk 1]
```diff
@@ -89,23 +95,31 @@ module.exports = async function(env) {
             new webpack.NoEmitOnErrorsPlugin(),
             new webpack.EnvironmentPlugin({
                 NODE_ENV: env.phase,
-                SERVICE_WORKER_FILE_NAME: 'sw.js'
+                // Blank the service worker file name to stop the app from
+                // attempting to register a service worker in index.js.
+                // Only register a service worker when in production or in the
+                // special case of debugging the service worker itself.
+                SERVICE_WORKER_FILE_NAME: JSON.stringify(
```
Comment (DrewML) [inline theme-frontend-venia/webpack.config.js]: Should this not be done from the `ServiceWorkerPlugin` automatically?

**Comment by zetlen** [comment]
This one is right in the middle of @jimbo's workflow, so we'll wait for him to approve before merging.

**Comment by DrewML** [comment]
@zetlen with @jimbo out for a week, would prefer to land this and we can handle the conflicts.

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento-research/venia-pwa-concept?pullRequest=68) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento-research/venia-pwa-concept?pullRequest=68) before we can accept your contribution.<br/>

**Pull request merged**